app/controllers/merchant_items_controller.rb

```diff
@@ -0,0 +1,27 @@
+class MerchantItemsController < ApplicationController
+  def index
+    @merchant = Merchant.find(params[:merchant_id])
+  end
+
+  def new
+    @merchant = Merchant.find(params[:merchant_id])
+  end
+
+  def create
+    @merchant = Merchant.find(params[:merchant_id])
+    item = @merchant.items.create(item_params)
+    if item.save
+      redirect_to "/merchants/#{@merchant.id}/items"
+    else
+      flash[:error] = item.errors.full_messages.to_sentence
+      render :new
+    end
+  end
+
+  private
+
+  def item_params
+    params.permit(:name,:description,:price,:inventory,:image)
+  end
+
+end
```

Comment:
This is great to have broken this logic out into a new controller.

Reply:
Ditto. So much cleaner